sdk/python/kfp/components/structures.py

```diff
@@ -616,11 +675,19 @@ def inputs_dict_from_components_dict(
             component_key = utils._COMPONENT_NAME_PREFIX + component_name
             parameters = components_dict[component_key].get(
                 'inputDefinitions', {}).get('parameters', {})
-            return {
+            parameter_input_specs = {
                 name: InputSpec.from_ir_parameter_dict(parameter_dict)
                 for name, parameter_dict in parameters.items()
             }
 
+            artifacts = components_dict[component_key].get(
```

Comment:
This is the v2 deserialization logic that was missing

Reply:
This is the main reason for this PR. The other changes are to make the serialization/deserialization logic consistent w.r.t. treatment of artifacts. This inconsistency made it very hard for me to test changes related to https://github.com/kubeflow/pipelines/pull/8172.